### PR TITLE
Remove blocker for JetBackup on Ubuntu

### PR DESCRIPTION
**Pull request by troglodyne**
Fix t/usage.t when noc_reccomendations file exists as well Fix bizzare errors in t/ test for ElevateScript that failed due to local configuration when running the testsuite during development.

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

File: elevate-cpanel [hunk 1]
```diff
@@ -7576,6 +7578,38 @@ deb https://wp-toolkit.plesk.com/cPanel/Ubuntu-22.04-x86_64/latest/thirdparty/ .
...
 
+    my %repo2filepart = (
+        map { my $dollar_underscore = $_; "jetapps-$dollar_underscore" => "repo.jetlicense.com_ubuntu_dists_jammy_$dollar_underscore" } qw{base plugins alpha beta edge rc release stable},
+    );
+
+    sub get_installed_pkgs_in_repo ( $self, @repos ) {
+        return unless scalar @repos;
+
+        my @apt_args = (APT_NON_INTERACTIVE_ARGS);
+        $self->ssystem_and_die( $apt, @apt_args, 'update' );
```
Comment (toddr) [inline elevate-cpanel:7589]: Where'd $apt come from?

**Review by toddr — CHANGES_REQUESTED** [review]
$apt?

**Comment by troglodyne** [comment]
It works because it's defined in the scope of Elevate::PkgMgr::APT, see line 31 of that package

**Comment by troglodyne** [comment]
rebased to main:HEAD

File: elevate-cpanel (hunk 1 above)
Comment (troglodyne) [reply elevate-cpanel:7589]: it's in the package scope

File: elevate-cpanel [hunk 2]
```diff
@@ -3074,6 +3074,13 @@ EOS
           : $repos->{'jetapps-beta'}   ? 'jetapps-beta'
           :                              'jetapps-stable';    # Just give up and choose stable if you can't guess.
         INFO("Jetbackup tier '$jetbackup_tier' detected. Not removing jetbackup. Will re-install it after elevate.");
+        if ( ref Elevate::PkgMgr::instance() eq 'Elevate::PkgMgr::APT' ) {
+            INFO("JB: CHECKING FOR /etc/apt/sources.list.d/$jetbackup_tier.disabled");
```
Comment (troglodyne) [inline elevate-cpanel:3078]: Bah, need to remove these debug lines

**Comment by troglodyne** [comment]
OK, finished dealing with the post-rebase headache, it should now be ONLY the changes I wanted.

I also invoked `$apt-get` in this case statically to reduce confusion.

File: lib/Elevate/Components/JetBackup.pm [hunk 3]
```diff
@@ -84,24 +91,12 @@ sub post_distro_upgrade ($self) {
...
-    return unless Cpanel::Pkgr::is_installed('jetbackup');
-    return if Elevate::OS::supports_jetbackup();
-
-    my $name = Elevate::OS::default_upgrade_to();
-    return $self->has_blocker( <<~"END" );
-    ELevate does not currently support JetBackup for upgrades of $name.
-    Support for JetBackup on $name will be added in a future version of ELevate.
-    END
-}
+sub _blocker_jetbackup_is_supported { return undef }
```
Comment (cPholloway) [inline lib/Elevate/Components/JetBackup.pm:99]: TRIVIAL: Does anything call this / could it just go away?

File: t/components-JetBackup.t [hunk 4]
```diff
@@ -84,23 +84,12 @@ my $jb   = $cpev->get_blocker('JetBackup');
 
     my $jb_mock = Test::MockModule->new('Elevate::Components::JetBackup');
 
-    for my $os ( 'cent', 'cloud' ) {
+    for my $os ( 'cent', 'cloud', 'ubuntu' ) {
         set_os_to($os);
 
         my $expected_target_os = Elevate::OS::upgrade_to_pretty_name();
         is( $jb->_blocker_jetbackup_is_supported(), undef, "JetBackup is supported for upgrades to $expected_target_os" );
     }
```
Comment (cPholloway) [inline t/components-JetBackup.t:92]: TRIVIAL: This whole block could go away now that Jetbackup is supported on all OSs

**Review by cPholloway — DISMISSED** [review]
Review LGTM / QA is still pending from me

File: lib/Elevate/Components/JetBackup.pm (hunk 3 above)
Comment (toddr) [reply lib/Elevate/Components/JetBackup.pm:99]: this will come up again on 8->9....

File: t/components-JetBackup.t (hunk 4 above)
Comment (toddr) [reply t/components-JetBackup.t:92]: this will come up again on 8->9....

**Review by toddr — DISMISSED** [review]

File: lib/Elevate/Components/JetBackup.pm (hunk 3 above)
Comment (troglodyne) [reply lib/Elevate/Components/JetBackup.pm:99]: Only the test for it

**Review by cPholloway — APPROVED** [review]

**Pull request merged**